barretenberg/cpp/src/barretenberg/vm2/constraining/testing/check_relation.hpp

```diff
@@ -1,23 +1,42 @@
 #pragma once
 
 #include <array>
+#include <cstdlib>
 #include <span>
 #include <stdexcept>
 
 #include "barretenberg/common/log.hpp"
+#include "barretenberg/honk/proof_system/logderivative_library.hpp"
+#include "barretenberg/relations/relation_parameters.hpp"
+#include "barretenberg/relations/relation_types.hpp"
+#include "barretenberg/vm2/constraining/polynomials.hpp"
 #include "barretenberg/vm2/tracegen/test_trace_container.hpp"
 
 namespace bb::avm2::constraining {
+namespace detail {
 
-template <typename Relation, typename Trace>
-void check_relation_internal(const Trace& trace, std::span<size_t> subrelations)
+const RelationParameters<FF>& get_test_params();
+
+template <typename Relation> constexpr bool subrelation_is_linearly_independent(size_t subrelation_index)
+{
+    if constexpr (HasSubrelationLinearlyIndependentMember<Relation>) {
+        return Relation::SUBRELATION_LINEARLY_INDEPENDENT[subrelation_index];
+    } else {
+        return true;
+    }
+}
+
+template <typename Relation, typename Trace, typename RowGetter>
```

Comment:
I needed to template how to get the row, because it works differently on traces and polynomials.